### PR TITLE
[AC-2568] Added invoices and transaction history endpoints. Added cursor paging for each

### PR DESCRIPTION
**Pull request by cturnbull-bitwarden**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/AC-2568

## 📔 Objective

Added invoices and transaction history endpoints. Added cursor paging for each

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by cyprain-okeke — APPROVED** [review]
Looks Good

**Review by amorask-bitwarden — COMMENTED** [review]
Just a few ⛏️s, but all looks good to me. Do you think it's worth adding some unit tests for the new service? Could also add controller tests, but the action methods are pretty simple so I'm not worried about them.

File: src/Core/Billing/Services/Implementations/PaymentHistoryService.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    IStripeAdapter stripeAdapter,
+    ITransactionRepository transactionRepository,
+    ILogger<PaymentHistoryService> logger) : IPaymentHistoryService
+{
+    public async Task<IEnumerable<BillingHistoryInfo.BillingInvoice>> GetInvoiceHistoryAsync(
+        ISubscriber subscriber,
+        int pageSize = 5,
+        string startAfter = null)
+    {
+        if (subscriber is null ||
```
[the remark this answers is not in the document]
Comment (cturnbull-bitwarden) [reply src/Core/Billing/Services/Implementations/PaymentHistoryService.cs]: Pushed!

File: src/Core/Billing/Services/Implementations/PaymentHistoryService.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            {
+                Customer = subscriber.GatewayCustomerId,
+                Subscription = subscriber.GatewaySubscriptionId,
+                Limit = pageSize,
+                StartingAfter = startAfter
+            });
+
+            return invoices.Select(invoice => new BillingHistoryInfo.BillingInvoice(invoice));
+        }
+        catch (StripeException exception)
```
[the remark this answers is not in the document]
Comment (cturnbull-bitwarden) [reply src/Core/Billing/Services/Implementations/PaymentHistoryService.cs]: Good call, I forgot about that. Pushed an update remove this try/catch

**Review by amorask-bitwarden — APPROVED** [review]

**Review by rkac-bw — DISMISSED** [review]
lgtm

**Review by amorask-bitwarden — APPROVED** [review]

**Review by rkac-bw — APPROVED** [review]
lgtm

**Pull request merged**